presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -1373,30 +1371,6 @@ protected void doTestMismatchSchemaTable(
             MaterializedResult result = readTable(transaction, tableHandle, columnHandles, session, TupleDomain.all(), OptionalInt.empty(), Optional.empty());
             assertEqualsIgnoreOrder(result.getMaterializedRows(), dataAfter.getMaterializedRows());
 
-            int filterCount = afterFilters.size();
-            for (int i = 0; i < filterCount; i++) {
-                RowExpression predicate = afterFilters.get(i);
-                ConnectorTableLayoutHandle layoutHandle = metadata.pushdownFilter(session, tableHandle, predicate, Optional.empty()).getLayout().getHandle();
```

Comment:
perhaps, make com.facebook.presto.hive.rule.HiveFilterPushdown#pushdownFilter @VisibleForTesting and use here?

Reply:
Okay I refactored a ton to make this static and testable - take a look and see if it makes sense.